### PR TITLE
More stylecop and documentation fixing

### PR DESCRIPTION
**Pull request by Ellerbach**
Following the previous PR, more stylecop and documentation fixing.
It should be good now for the GenICam project. It's just code style fixing, few other things.
I hope this will be reflect in the nuget to get a great documentation :-)

### CODE REVIEW

**Comment by Touseefelahi** [comment]
Sure thing, I'll be updating the documentation part as well.
Just out of curiosity are you using some auto documentation utility?

**Review by Touseefelahi — APPROVED** [review]

**Comment by Ellerbach** [comment]
> @Ellerbach Just out of curiosity are you using some auto documentation utility?

I'm huge fan of DocFx to build automatically the documentation :-) Check:

* this starter guide: https://github.com/mtirionMSFT/DocFxQuickStart
* the additional tools: https://github.com/Ellerbach/docfx-companion-tools
* and a full example with a lot of complex projects to build in the real life: https://github.com/nanoframework/nanoframework.github.io, most is in the pipeline https://github.com/nanoframework/nanoframework.github.io/blob/pages-source/azure-pipelines.yml and in the toc and other specific files and the result here: https://docs.nanoframework.net/ with all the API of all system related repo here: https://docs.nanoframework.net/api/index.html And everything is fully build automatically, no human effort at all and it's like 80+ repos cloned, build with some more specific content.

Happy to help for your usage! The starter kit should contain what you need and quite some more. Ther should be a git workflow example as well.

**Comment by Ellerbach** [comment]
> @Ellerbach Just out of curiosity are you using some auto documentation utility?

And if I miss understood the question, if it's about commenting all the code, no, it's fully done by hand! That say, when using StyleCop and Visual Studio (the real one even with the Community free version, not VS Code), you get some help with alt + enter or right click with more actions or the small menu which appear and can correct some of the docs elements directly like setting the standard texts, adjusting the brackets, things like this.
In other words, it's still a lot of manual work. Now, you made me gain some time as I did not have to rewrite an equivalent library, and I have to say, documentation and proper indentation in the code is important for me when working, debugging and trying to understand the code. So I gave you back something really clean in terms of presentation and documentation for this time gained :-) But yes, I spend hours on fixing all!

**Comment by Touseefelahi** [comment]
@Ellerbach Thank you, I really appreciate your work
